packages/test/test-version-utils/README.md

```diff
@@ -1,26 +1,137 @@
+to enable compat testing easily in the future just by changing the `describe*`.
+
+### Legacy version defaults and installation
+
+By default, N-1, N-2, and LTS (hard coded) test variants are generated.  The versions can be specified using command
+line (see below) to run the test against any two versions. This package includes a `mocha` global hook that will
+install legacy packages at the beginning of the package based on the `compatVersion` settings.
+
+## Command line options
+
+Tests using the compat `describe*` will be controllable using the command line options when running mocha on
+driver selection, versions for compat testing, and compat kind combinations.
+
+```text
+--compatKind <CompatKind> - filter to the compat variant. See above table. Can specify multiple times.
+                                Default: undefined (no filter)
+--compatVersion <version> - specify the old version. Relative (to base) or specific version. Can specify multiple times.
+                                Default: -1, -2, LTS (hard coded in src/compatConfig.ts)
+--baseVersion <version>   - specify the base (new) version. Allow the test to run against any version combinations.
+                                Default: <current> (same version of the test package)
+--driver <driverType>     - <driverType> = "tinylicious" | "t9s" | "routerlicious" | "r11s" | "odsp" | "local"
+                                Default: "local"
+--r11sEndpointName <name> - Determine the environment variable name to look for r11s service information to run against
+                                Default: "r11s"
+--tenantIndex <number>    - The index into the tenant list to look for odsp server information to run against
+                                Default: 0
+--reinstall               - Force reinstall any required versions.  Default: reuse if a version is already installed.
+```
+
+`baseVersion` is a semver or a semver range.
+
+- If it is a semver range, the latest in that range will be picked.
+
+`compatVersion` can be a semver or semver range or an integer apply to the minor version relative to `baseVersion`
+
+- If it is a semver range, the latest in that range will be picked.
+- If it is an integer, the value will be apply to the minor version of the `baseVersion`, and create a `^` range
+including prerelease versions. The latest in that range will be picked.
+  - i.e. if `baseVersion` is `1.2.3`, and `compatVersion` is `-1`, the resulting range
+will be `^1.1.3-0`
+
+
+We also accept some of the flags via environment variables.
+
+| Environment Variables         | Command line options |
+| ----------------------------- | -------------------- |
+| fluid__test__compatKind       | --compatKind         |
+| fluid__test__compatVersion    | --compatVersion      |
+| fluid__test__driver           | --driver             |
+| fluid__test__r11sEndpointName | --r11sEndpointName   |
+| fluid__test__baseVersion      | --baseVersion        |
+
+## Advanced usage
```

Comment:
I think a short "sample usage" code block like this would be a great supplement to the words you wrote.

```
import { ensurePackageInstalled, getVersionedTestObjectProvider } from ....

await ensurePackageInstalled("0.50.0");

// get whole test provider
await getVersionedTestObjectProvider("0.50.0");

// get just the ContainerRuntime APIs
await getContainerRuntimeApi("0.50.0");
```

And doc comments on the functions you mention here would be super awesome.

Reply:
I am trying to not dwell on/expand this part yet, as it is advanced usage, and it hasn't been proven out fully.